### PR TITLE
Add new AKS ManagedCluster version

### LINKED ISSUE #2620: Supporting creating overlay managed cluster
To create an overlay cluster, we need to set field `networkPluginMode` to overlay (see [doc](https://learn.microsoft.com/en-us/python/api/azure-mgmt-containerservice/azure.mgmt.containerservice.v2022_06_02_preview.models.containerservicenetworkprofile?view=azure-python#variables)). Currently the field `networkPluginMode` is not exposed within the CRD of managedCluster.

### PR DESCRIPTION
**Pull request by matthchr**
Closes #2620 

**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains tests

### CODE REVIEW

File: v2/samples/containerservice/v1beta/v1beta20230101_managedcluster.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+apiVersion: containerservice.azure.com/v1beta20230101
```
Comment (super-harsh) [inline v2/samples/containerservice/v1beta/v1beta20230101_managedcluster.yaml:1]: Consider moving different versions into separate dirs as we discussed earlier for `dbforpostgresql`

https://github.com/Azure/azure-service-operator/tree/main/v2/samples/dbforpostgresql

**Review by theunrepentantgeek — APPROVED** [review]

File: v2/samples/containerservice/v1beta/v1beta20230101_managedcluster.yaml (hunk 1 above)
Comment (matthchr) [reply v2/samples/containerservice/v1beta/v1beta20230101_managedcluster.yaml]: Done

**Comment by matthchr** [comment]
> [controller:install-helm] Error: create: failed to create: Secret "sh.helm.release.v1.asov2.v1" is invalid: data: Too long: must have at most 1048576 bytes

Looks like we have to deal with this first.

**Review by super-harsh — APPROVED** [review]

**Pull request merged**